apps/src/lib/util/speech.js

```diff
@@ -22,38 +20,149 @@ export function textToSpeech(
   region,
   appLanguages
 ) {
-  const speechConfig = SpeechConfig.fromAuthorizationToken(token, region);
-  speechConfig.speechSynthesisOutputFormat =
-    SpeechSynthesisOutputFormat.Audio16Khz32KBitRateMonoMp3;
-  let voice =
-    (appLanguages[language] && appLanguages[language][gender]) ||
-    appLanguages['English']['female'];
-  const synthesizer = new SpeechSynthesizer(
-    speechConfig,
-    undefined /* AudioConfig */
-  );
-  let ssml = `<speak version="1.0" xmlns="https://www.w3.org/2001/10/synthesis" xml:lang="en-US"><voice name="${voice}">${text}</voice></speak>`;
-  synthesizer.speakSsmlAsync(
-    ssml,
-    result => {
-      // There is no way to make ajax requests from html on the filesystem.  So
-      // the only way to play sounds is using HTML5. This scenario happens when
-      // students export their apps and run them offline. At this point, their
-      // uploaded sound files are exported as well, which means varnish is not
-      // an issue.
-      const forceHTML5 = window.location.protocol === 'file:';
-      Sounds.getSingleton().playBytes(result.audioData, {
+  if (text.length > 749) {
+    text = text.slice(0, 750);
+    outputWarning(i18n.textToSpeechTruncation());
+  }
+  let voice = appLanguages['English']['female'];
+  let languageCode = 'en-US';
+  if (appLanguages[language] && appLanguages[language][gender]) {
+    voice = appLanguages[language][gender];
+    languageCode = appLanguages[language]['languageCode'];
+  } else {
+    language = 'English';
+    gender = 'female';
+  }
```

Comment:
nit: set these as defaults in the same way you set `let voice = ` on line 27

Reply:
I don't think I can since language and gender are parameters passed in to the function, I'm just setting them to English and female if they were invalid

Reply:
woops, you're right -- forget that!